c_example_shlib.c

```diff
@@ -0,0 +1,58 @@
+#include <stdio.h>
+#include <dlfcn.h>
+#include <stdint.h>
+#include <sched.h>
+
+
+/* Library functions declaration (instead of .h file) */
+int pcm_c_build_core_event(uint8_t, const char *);
+int pcm_c_init();
+void pcm_c_start();
+void pcm_c_stop();
+uint64_t pcm_c_get_cycles(uint32_t);
+uint64_t pcm_c_get_instr(uint32_t);
+uint64_t pcm_c_get_core_event(uint32_t, uint32_t);
+
+
+int main(int argc, const char *argv[])
+{
+	int i,a[100],b[100],c[100];
+	int lcore_id;
+
+	switch(argc-1)
+	{
+		case 4:
+			pcm_c_build_core_event(3,argv[3]);
+		case 3:	
+			pcm_c_build_core_event(2,argv[2]);
+		case 2:
+			pcm_c_build_core_event(1,argv[2]);
+		case 1:
+			pcm_c_build_core_event(0,argv[1]);
+		case 0:
+			break;
+		default:
+			printf("Number of arguments are too many! exit...\n");
+			return -2;
+	}
+
+	pcm_c_init();
+	pcm_c_start();
+	for(i=0;i<10000;i++)
+            c[i%100] = 4 * a[i%100] + b[i%100];
+	pcm_c_stop();
+
+	lcore_id = sched_getcpu();
+	printf("C:%lu I:%lu, IPC:%3.2f\n",
+		pcm_c_get_cycles(lcore_id),
+		pcm_c_get_instr(lcore_id),
+		(double)pcm_c_get_instr(lcore_id)/pcm_c_get_cycles(lcore_id));
+	printf("CPU%d E0: %lu, E1: %lu, E2: %lu, E3: %lu\n",
+		lcore_id,
+		pcm_c_get_core_event(lcore_id,0),
+		pcm_c_get_core_event(lcore_id,1),
+		pcm_c_get_core_event(lcore_id,2),
+		pcm_c_get_core_event(lcore_id,3));
+
+	return 0;
+}
```

Comment:
there is a lot of code duplication compared to c_example.c

It would be better not to introduce the new file but instead to have a few if defs (PCM_SHARED_LIB) for your changes and then have an -DPCM_SHARED_LIB in the new c_example_shlib.x target compilation.

Reply:
Thank you for your feedback.  
I've updated the pull request.